cmd/server/assets/login/_loginscripts.html

```diff
@@ -155,67 +155,289 @@
     if (pwd.length < {{.requirements.Length}}) {
       $lenReq.find("#icon").attr("class", errClass)
       $lenReq.addClass("text-danger");
+      $lenReq.removeClass("text-muted");
       valid = false;
     } else {
       $lenReq.find("#icon").attr("class", checkClass)
       $lenReq.addClass("text-muted");
+      $lenReq.removeClass("text-danger");
     }
     {{end}}
 
     {{if gt .requirements.Uppercase 0}}
     if (upper < {{.requirements.Uppercase}}) {
       $upperReq.find("#icon").attr("class", errClass);
       $upperReq.addClass("text-danger");
+      $upperReq.removeClass("text-muted");
       valid = false;
     } else {
       $upperReq.find("#icon").attr("class", checkClass);
       $upperReq.addClass("text-muted");
+      $upperReq.removeClass("text-danger");
     }
     {{end}}
 
     {{if gt .requirements.Lowercase 0}}
     if (lower < {{.requirements.Lowercase}}) {
       $lowerReq.find("#icon").attr("class", errClass);
       $lowerReq.addClass("text-danger");
+      $lowerReq.removeClass("text-muted");
       valid = false;
     } else {
       $lowerReq.find("#icon").attr("class", checkClass);
       $lowerReq.addClass("text-muted");
+      $lowerReq.removeClass("text-danger");
     }
     {{end}}
 
     {{if gt .requirements.Number 0}}
     if (digit < {{.requirements.Number}}) {
       $numReq.find("#icon").attr("class", errClass);
       $numReq.addClass("text-danger");
+      $numReq.removeClass("text-muted");
       valid = false;
     } else {
       $numReq.find("#icon").attr("class", checkClass);
       $numReq.addClass("text-muted");
+      $numReq.removeClass("text-danger");
     }
     {{end}}
 
     {{if gt .requirements.Special 0}}
     if (special < {{.requirements.Special}}) {
       $specialReq.find("#icon").attr("class", errClass);
       $specialReq.addClass("text-danger");
+      $specialReq.removeClass("text-muted");
       valid = false;
     } else {
       $specialReq.find("#icon").attr("class", checkClass);
       $specialReq.addClass("text-muted");
+      $specialReq.removeClass("text-danger");
     }
     {{end}}
 
     if (pwd != $retype.val()) {
       $retyped.find("#icon").attr("class", errClass);
       $retyped.addClass("text-danger");
+      $retyped.removeClass("text-muted");
       valid = false;
     } else {
       $retyped.find("#icon").attr("class", checkClass);
       $retyped.addClass("text-muted");
+      $retyped.removeClass("text-danger");
     }
 
     {{end}}
     return valid;
   }
 {{end}}
+
+{{define "login/login-js"}}
```

Comment:
Is it feasible to move this to application.js while we're moving things around? Anything that depends on server-side information we can populate in the HTML with hidden fields and then pull them in the js.

Reply:
move password complexity into application.js and moved server-side stuff into a js struct to use as a param.

still looking at the rest, but I get a little concerned by the complexity of this refactor